### ISSUE TITLE
Add support for "listing" column style 

### ISSUE THREAD

**Issue by benignbala** [asciidoctor/asciidoctor#1473]
Just like we have support for a,e,h etc. as column styles, we should have a listing style that accepts an optional second parameter which can be used with source code highlighers. For eg. if we assume 'g' with be the column style to denote "listing" type,

|===
g,python|
import os
print (os.uname())
|===

Should be a short cut for
|===
a|
[source,python]
import os
print (os.uname())
|===

**Comment by mojavelinux** [comment]
I like this idea. Seems like a very natural fit.

I don't think "g" is the right choice though. There's already a literal style for table cells, "l", which goes mostly unused. I wonder if we can somehow build on that.

```
|===
l,python|
import os
print (os.uname())
|===
```

We don't currently have a reserved space in the syntax for attributes on a table cell, but I suppose we can try to find where they belong.

**Comment by benignbala** [comment]
Shall I try to add support and send in a PR ? Thanks

**Comment by mojavelinux** [comment]
I think we should work more on the design before jumping into implementation.

**Comment by benignbala** [comment]
Agreed :) 

BTW, I was suggesting a separate 'g' for source listing so that the processing is easier(to look for a language attribute too). But I agree that source listing is just a special case of literals and we can reuse 'l' for this too.

**Comment by benignbala** [comment]
Is there a way to enable some sort of log to get into what the  Parser is doing ? What I am looking for is, when asciidoctor converts a source doc into some format, as the Parser reads each line, is there a way to log what is currently being consumed/processed by the next_table method ?

**Comment by mojavelinux** [comment]
Nope. A log would slow down the parser tremendously (one of the downsides of Ruby is that every processing instruction costs time). I tend to just put in statements that print information from the reader. However, a more sound approach is to use RubyMine so that you can use the Ruby debugger. (we have a license for RubyMine if you need it).

**Comment by benignbala** [comment]
Thanks for the info. For now, I'll try adding print statements. I'll request RubyMine license if I am still lost :)

**Comment by mojavelinux** [comment]
:+1:

**Comment by benignbala** [comment]
I did some debugging and here is what I think we can do. We are using a regex(CellSpecStartRx) to identify cell style specifications. So, we can try

1) Adding a SecondarySepcification regex (which is optional) and processing the secondary only if the primary spec is 'l' , or
2) Process the second attribute as well with the same regex, and then split by comma to see if there are multiple specs.

In either case, we should process the "listing" style as a separate block.

**Comment by mojavelinux** [comment]
> processing the secondary only if the primary spec is 'l' 

I don't like the idea of being that specific.

I've given this a lot of thought and considered the proposed changes and I've decided that it takes the AsciiDoc syntax in a direction that opens more questions than it addresses. We'd effectively be turning the cell spec into an attributes list, and that opens up a lot of potential for parsing something we shouldn't.

Don't despair, I come bearing a new proposal!

If we consider:

```
|===
l,python|
import os
print (os.uname())
|===
```

vs

```
|===
a|[source,python]
import os
print (os.uname())
```
|===

We're really only saving 8 characters, most of which are reducing "source" to "l".

What I am willing to do is auto-detect an AsciiDoc table cell based on the first line of content. That would alleviate the need of having to type the "a".

```
|===
|[source,python]
import os
print (os.uname())
|===
```

We're also talking about finding a way to make "[source,python]" more abbreviated. Perhaps allowing "src" as an alternative to "source".

```
|===
|[src,python]
import os
print (os.uname())
|===
```

**Comment by mojavelinux** [comment]
I have another alternate proposal. "l" is already reserved for the literal cell type. We could augment this to allow a language, similar to a fenced code block, as follows:

```
|===
l|python
import os
print (os.uname())
|===
```

This slightly runs the risk of parsing text that was meant to be literal.

**Comment by mojavelinux** [comment]
...though if we are worried about conflicting with literal, I suppose we could introduce "c" for code.

```
|===
c|python
import os
print (os.uname())
|===
```

or allow multiple letters, which I'd rather avoid.

```
|===
src|python
import os
print (os.uname())
|===
```

**Comment by benignbala** [comment]
Just back from my vacations, Sorry about the delayed response :(

I like the c|python style. Looks more natural when we type. multi-letters are ok, but the c|<lang> is simpler.
